### DISCUSSION TITLE
[Table] Access public property in getTableFilters()

### DISCUSSION THREAD

**Discussion by wit3** [filamentphp/filament#701] [Help]
Hi,
I'm try to create a filter for my tabled based on a property passed in Mount method.

When i defined filter using this method

```
protected function getTableFilters(): array
{
  return [
      SelectFilter::make('status')
      ->options($this->fooArray)
  ];
}
```
An error occured 

`Typed property App\Http\Livewire\Tables\AccountList::$cachedTableFilters must not be accessed before initialization`

or if i try to use `$this->team_id` in `getTableFilters()` method, the value is null

PS: `$this->fooArray` is a computedProperty
```
public function getFooArrayProperty()
{
    return User::query()
    ->where('team_id',$this->team_id)
    ->pluck('id', name')
    ->toArray();
}
```

Why i cannot access to my public property when defined filter?

**Comment by danharrin** [comment]
Can you send a screenshot of the stack trace for that error?

**Comment by wit3** [comment]
Suspect that the problem is another, if i try to access a property in getTableFitlers method is always null, because the mount() method execute after the getTableFilters in InteractsWithTable trait.

**Reply by wit3** [reply]
This problem is perhaps solved by moving in Trait `InteractsWithTable`

from method `bootInteractsWithTable`

```
 $this->cacheTableFilters();
 $this->getTableFiltersForm()->fill();
```

to `bootedInteractsWithTable` in 

```
    public function bootedInteractsWithTable(): void
    {
        $this->cacheTableFilters();
        $this->getTableFiltersForm()->fill();
    }
```

if you do this, all public proprieties setted in `mount` method from component now is accessible, and that resolve the problem ( create filter based on query properties )

**Reply by danharrin** [reply]
Awesome! Please PR to rename that method, I also think that columns and actions should be booted after `mount()`.
